libbeat/autodiscover/providers/kubernetes/config.go

```diff
@@ -24,7 +24,8 @@ type Config struct {
 
 func defaultConfig() *Config {
 	return &Config{
-		InCluster:  true,
-		SyncPeriod: 1 * time.Second,
+		InCluster:      true,
+		SyncPeriod:     1 * time.Second,
+		CleanupTimeout: 60 * time.Second,
```

Comment:
I understand to provide a default values for the cleanup, but do we need to change the locking strategy to fix the issue?

Reply:
ah nevermind I understand now :)

Reply:
Not really, but I think that slipped my review, locking while sending to the bus is not needed so I moved it outside. I also added a debug line.